packages/ai/src/ui/process-ui-message-stream.ts

```diff
@@ -350,7 +351,7 @@ export function processUIMessageStream<UI_MESSAGE extends UIMessage>({
               // Skip calling onToolCall for provider-executed tools since they are already executed
               if (onToolCall && !chunk.providerExecuted) {
                 const result = await onToolCall({
-                  toolCall: chunk,
+                  toolCall: chunk as InferUIMessageToolCall<UI_MESSAGE>,
```

Comment:
The `UIMessageChunk` type (passed to the `TransformStream`) doesn’t take any arguments and is used in several places across the codebase. Because of this, I opted for a simpler approach and just did a basic type cast.